File: ProjectTracker20vb/www/PTWeb/MasterPage.master.vb
Option Strict On

Partial Class MasterPage
    Inherits System.Web.UI.MasterPage

  Protected Sub Page_Load( _
    ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load

    PageTitle.Text = Page.Title

  End Sub

  Protected Sub LoginStatus1_LoggingOut( _
    ByVal sender As Object, _
    ByVal e As System.Web.UI.WebControls.LoginCancelEventArgs) _
    Handles LoginStatus1.LoggingOut

    ProjectTracker.Library.Security.PTPrincipal.Logout()
    Session("CslaPrincipal") = Csla.ApplicationContext.User
    System.Web.Security.FormsAuthentication.SignOut()

  End Sub

End Class

File: Samples/NET/cs/ProjectTracker/PTWebForms/MasterPage.master.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace PTWebForms
{
  public partial class MasterPage : System.Web.UI.MasterPage
  {
    protected void Page_Load(object sender, EventArgs e)
    {
      PageTitle.Text = Page.Title;
    }

    protected void LoginStatus1_LoggingOut(
      object sender, LoginCancelEventArgs e)
    {
      ProjectTracker.Library.Security.PTPrincipal.Logout();
      Session["CslaPrincipal"] =
        Csla.ApplicationContext.User;
      System.Web.Security.FormsAuthentication.SignOut();
    }
  }
}
